crates/wasmtime/src/module.rs

```diff
@@ -332,8 +336,28 @@ impl Module {
         }
     }
 
-    /// Compiles a binary-encoded WebAssembly module to an artifact usable by
-    /// Wasmtime.
+    /// Creates a new WebAssembly `Module` from the contents of the given
+    /// `file` on disk, but with assumptions that the file is 'sane'.
+    /// In other words, the file should be a binary- or text-format WebAssembly
+    /// module or a precompiled version of it generated by wasmtime.
+    ///
+    /// # Unsafety
+    ///
+    /// This function is marked as `unsafe` as it relies on [`Module::deserialize`].
+    #[cfg(compiler)]
+    #[cfg_attr(nightlydoc, doc(cfg(feature = "cranelift")))] // see build.rs
+    pub unsafe fn from_trusted_file(engine: &Engine, file: impl AsRef<Path>) -> Result<Module> {
+        let mmap = MmapVec::from_file(file.as_ref())?;
+        if &mmap[0..4] == b"\x7fELF" {
+            return SerializedModule::from_mmap(mmap, &engine.config().module_version)?
+                .into_module(engine);
+        }
+
+        Module::new(engine, &*mmap)
```

Comment:
Out of curiosity, is the `*` necessary here? I think if you write `&mmap`, Rust will still figure out that you want the dereferenced type. But I think the `&` is necessary either way.

Reply:
It needs a type implementing `AsRef<[u8]>` as argument. `&MmapVec` doesn't implement this. `MmapVec` only implements `Deref` and `DerefMut`.